src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -59,32 +60,56 @@ class DiscreteDistribution
    *    can have.
    */
   DiscreteDistribution(const size_t numObservations) :
-      probabilities(arma::ones<arma::vec>(numObservations) / numObservations)
+     probabilities(std::vector<arma::vec>(1, arma::ones<arma::vec>(numObservations)/numObservations))
   { /* nothing to do */ }
 
   /**
-   * Define the discrete distribution as having the given probabilities for each
+   * Define the multidimensional discrete distribution as having numObservations possible
+   * observations.  The probability in each state will be set to (1 /
+   * numObservations of each dimension).
+   *
+   * @param numObservations Number of possible observations this distribution
+   *    can have.
+   */
+  DiscreteDistribution(const arma::vec& numObservations)
```

Comment:
I realized, we can make this `arma::Col<size_t>` not `arma::vec` and that would be cleaner, but I can do that after merge if you like.  I don't want to have an endless stream of comments so that this never gets merged. :)